packages/amplify-provider-awscloudformation/src/initializer.ts

```diff
@@ -92,26 +92,41 @@ export const run = async (context: $TSContext): Promise<void> => {
       },
     };
 
+    let projectInitialized = false;
+    let overrideFilePath = '';
     try {
       const backendDir = pathManager.getBackendDirPath();
-      const overrideFilePath = path.join(backendDir, 'awscloudformation', 'build', 'override.js');
-      const overrideCode: string = await fs.readFile(overrideFilePath, 'utf-8');
-      if (overrideCode) {
-        const sandboxNode = new vm.NodeVM({
-          console: 'inherit',
-          timeout: 5000,
-          sandbox: {},
-          require: {
-            context: 'sandbox',
-            builtin: ['path'],
-            external: true,
-          },
-        });
-        await sandboxNode.run(overrideCode).override(configuration);
-      }
+      overrideFilePath = path.join(backendDir, 'awscloudformation', 'build', 'override.js');
+      projectInitialized = true;
     } catch (e) {
-      // TODO: this always throws, and likely needs re-written
-      printer.debug(`Unable to apply auth role overrides: ${e.message}`);
+      // project not initialized
+    }
+    // when a new environment is created, init is called, and we must apply overrides for the auth/unauth roles
+    // not sure why we only do this for auth...
```

Comment:
This code is not related to the `auth` category specifically. The `auth` and `unauth` role names are created on `init` for legacy implementation reasons. I think this comment adds more confusion than anything.

```suggestion
```